core/src/main/java/org/infinispan/protostream/TagHandler.java

```diff
@@ -13,8 +15,10 @@
 
    /**
     * Start of top level message. Do whatever required init here.
+    *
+    * @param descriptor The {@link GenericDescriptor} for the top level Enum or Message. For primitives it is null.
     */
-   default void onStart() {
+   default void onStart(AnnotatedDescriptor descriptor) {
```

Comment:
My concern is whether source compatibility would be maintained when migrating from 7.1 to 7.2, but since this API is internal, users should not be affected, unless they are using protostream to other ends